nova/tests/unit/scheduler/filters/test_shard_filter.py

```diff
@@ -234,3 +237,62 @@ def test_non_vmware_spec(self, mock_is_non_vmware_spec):
 
         self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
         mock_is_non_vmware_spec.assert_called_once_with(spec_obj)
+
+    def test_ignores_hana_by_trait(self):
+        extra_specs = {'trait:CUSTOM_HANA_EXCLUSIVE_HOST': 'required'}
+        self._shard_filter_ignoring_hana(True, extra_specs)
+
+    def test_ignores_hana_by_memory(self):
+        CONF.set_override('bigvm_mb', 8192)
+        CONF.set_override('hana_detection_strategy', 'memory_mb',
+                          'filter_scheduler')
+        extra_specs = {}
+        self._shard_filter_ignoring_hana(True,
+                                         extra_specs=extra_specs,
+                                         memory_mb=8192)
```

Comment:
There's a test missing for the opposite: `memory_mb=CONF.bigvm_mb - 1` with `False`.

Reply:
No need to change it, but I do like it more if we write down the intended values i.e. `CONF.bigvm_mb - 1` instead of `8191` because the latter isn't obvious without looking at multiple lines before to see that `bigvm_mb` was set to `8192`.